telegram/client.go

```diff
@@ -254,11 +254,14 @@ func (c *Client) runUntilRestart(ctx context.Context) error {
 	g := tdsync.NewCancellableGroup(ctx)
 	g.Go(c.conn.Run)
 	g.Go(func(groupCtx context.Context) error {
-		_, err := c.tg.HelpGetConfig(ctx)
-		if err != nil {
-			c.log.Warn("Get config failed", zap.Error(err))
+		self, err := c.Self(ctx)
+		// Ignore unauthorized errors.
+		if err != nil && !unauthorized(err) {
+			c.log.Warn("Got error on self", zap.Error(err))
+			return nil
```

Comment:
Should we return nil here or err?

Reply:
Returned error will cause reconnect, I see no reason why RPC error(e.g. `FLOOD_WAIT`) should cause reconnection.